### ISSUE TITLE
Add build and test job in CI pipeline

### ISSUE THREAD

**Issue by Aadesh-Baral** [Aadesh-Baral/OSMLocalizer#86]
Add a job on cirlce ci that will be triggered on every pull request.
This job should test frontend build and backend image build for now.
This should also check the code formatting.
Later we need to add tests for both backend and frontend in the job.

**Comment by sagargg** [comment]
We could use GitHub Actions also, GitHub provides free  2000Mins  ( or 33.33 hours)  CI/CD time for individuals and organizations.   https://github.com/pricing
But, It appears that there are currently no tests written at the moment.

**Comment by Aadesh-Baral** [comment]
Yes, we do not currently have any tests, but in order to maintain the project in the long run, test cases will be required. However, for the time being, we can perform tests on frontend builds and backend code linting. Since we have been utilizing Circle CI for CI purposes, it would be beneficial if we could use it for this work.
